src/content/get-started/ch2-add-commands.md

```diff
@@ -0,0 +1,397 @@
+      }
+    }
+    ```
+
+1.  **Understand the `if/else` structure and variables:** You've now built an
+    `if`, `else if`, and `else` structure for conditional execution.
+
+    * `arguments.isNotEmpty` checks if any command-line arguments were
+        provided.
+    * `arguments.first` accesses the very first argument, which we're using as
+        our command.
+    * Notice the `version` variable is declared as a `const`. This means its
+        value is known at compile time and you cannot change it during runtime.
+    * `arguments`, on the other hand, is a regular (non-constant) variable
+        because its content can change based on user input.
+
+    Test the `help` command: Run your application with the help argument:
+
+    ```bash
+    dart bin/cli.dart help
+    ```
+
+    You should see the usage information printed.
+
+    Also, try running it without any arguments:
+
+    ```bash
+    dart bin/cli.dart
+    ```
+
+    It should now also display the usage information.
+
+    :::note
+    At this point, any command we haven't defined will also print usage
+    information. This is expected behavior for now.
+    :::
+
+### Task 2: Implement the search command
+
+Next, implement a basic `search` command that takes an article title as
+input. As you build this functionality, you'll work with `List` manipulation,
+null checks, and string interpolation.
+
+1.  **Integrate the `search` command into `main`:** First, modify your `main`
+    function in `cli/bin/cli.dart` to include an `else if` branch that handles
+    the `search` command. For now, we'll just print a placeholder message.
+
+    ```dart
+    void main(List<String> arguments) {
+      if (arguments.isNotEmpty && arguments.first == 'version') {
+        print('Dartpedia CLI version $version');
+      } else if (arguments.isNotEmpty && arguments.first == 'help') {
+        printUsage();
+      } else if (arguments.isNotEmpty && arguments.first == 'search') {
+        // Add this new block:
+        print('Search command recognized!');
+      } else {
+        printUsage();
+      }
+    }
+    ```
+
+1.  **Test the new command:** Run your application with the `search`
+    command:
+
+    ```bash
+    dart bin/cli.dart search
+    ```
+
+    You should see:
+
+    ```bash
+    Search command recognized!
+    ```
+
+1.  **Define the `runApp` function:** The `search` command will eventually run
+    the core logic of your application by calling a function called `runApp`.
+    For now, it will print the arguments passed into search. Place this new
+    function below `main`.
+
+    ```dart
+    // ... (your existing main function)
+
+    void runApp(List<String>? arguments) { // Add this new function and add ? to arguments type
+      print('runApp received arguments: $arguments');
+    }
+
+    // ... (your existing printUsage() function)
+    ```
+
+    * The `List<String>? arguments` type signature means that the `arguments`
```

Comment:
I think you should preface these bulleted lists with something like this:

Highlights from the preceding code: